src/matUtils/uncertainty.cpp

```diff
@@ -0,0 +1,357 @@
+#include "uncertainty.hpp"
+
+std::vector<MAT::Node*> get_common_nodes (std::vector<std::vector<MAT::Node*>> nodepaths) {
+    //to identify common nodes, perform pairwise set intersections repeatedly for all path node vectors
+    std::vector<MAT::Node*> common_nodes = nodepaths[0];
+    std::sort(common_nodes.begin(), common_nodes.end()); //needs to be sorted for intersection. These are actually vectors of node POINTERS, so this should be fine.

```

Comment:
@jmcbroome I usually prefer not having very long comments on the same line as the code. They should go on top.